src/audio/selector/selector.c

```diff
@@ -579,7 +579,12 @@ static void build_config(struct comp_data *cd, struct module_config *cfg)
 	enum sof_ipc_frame __sparse_cache frame_fmt, valid_fmt;
 	int i;
 
-	frame_fmt = cfg->base_cfg.audio_fmt.depth;
+	/*
+	 * When compiling with CONFIG_IPC_MAJOR_4 depth is of
+	 * enum ipc4_bit_depth type and requires a cast to numeric type
+	 * before recasting to other enum.
+	 */
+	frame_fmt = (uint32_t)cfg->base_cfg.audio_fmt.depth;
```

Comment:
Why do we need to have a conversion between different enums in the first place?

Reply:
@aborisovich which enums are impacted here ? Sounds like we may need an alignment.

Reply:
Well, when `sof/lib/dai.h` is included, depth is a `uint32_t` type member of `dai_plat_fifo_data` struct.
However when we compile with IPC4 config option, header `src\include\ipc4\base-config.h` is uder and depth is `enum ipc4_bit_depth` type what causes the warning:

> implicit conversion from enumeration type 'enum ipc4_bit_depth' to different enumeration type 'enum sof_ipc_frame' [-Wenum-conversion]

Reply:
Yes, as @paulstelian97 indicated this looks like a serious bug.
Enums are implicitly cast to numerical values so what we do here,
is trying to assign one of the values from this set { 8, 16, 24, 32, 64, 65 } (values of enumerators of `enum ipc4_bit_depth`) as enumerator with such index to `enum sof_ipc_frame` which has enumerator values { 0, 1, 2, 3, 4 }.
I'm quite positive this is undefined behavior in C.